tests/integration/models/os.spec.ts

```diff
@@ -664,6 +665,33 @@ describe('OS model', function () {
 					});
 			});
 
+			it('should be able to configure v2 image with a provisioning key name', async function () {
+				const configOptions = {
+					appUpdatePollInterval: 72,
+					network: 'wifi' as const,
+					wifiKey: 'foobar',
+					wifiSsid: 'foobarbaz',
+					ip: '1.2.3.4',
+					gateway: '5.6.7.8',
+					netmask: '9.10.11.12',
+					version: '2.11.8+rev1.prod',
+					provisioningKeyName: 'provision-key',
+				};
+
+				await balena.models.os.getConfig(ctx.application.id, configOptions);
+
+				const provisioningKeys =
+					await balena.models.apiKey.getProvisioningApiKeysByApplication(
+						ctx.application.id,
+					);
+
+				expect(provisioningKeys).to.be.an('array');
+				expect(provisioningKeys).to.have.length(1);
```

Comment:
This still failing.
For sure there are more than 1 keys at this point in the test suite.
Why don't you just do it just like in the API PR?
See: https://github.com/balena-io/balena-sdk/pull/1107/files#diff-6e5edef2fac65498ec08862f218a130979112a22de7b03b6405eda18c16eb445R688-R702

Reply:
I know - I was looking to move away from depending on order and checking `[0]` index as well. Updated to generic `find` operation now.